### PR TITLE
[Breaking] fix(Flags): consolidation and help text improvements

### PR DESCRIPTION
**Pull request by karlmcguire**
DGRAPH-3172

| Old Name | New Name | Type | Subcommands |
|--------------:|----------------:|:-------:|:--------------------|
| | **`--telemetry`** | | | |
| `telemetry` | `reports` | bool | alpha, zero |
| `enable_sentry` | `sentry` | bool | alpha, zero |
| | **`--limit`** | | |
| `mutations` | `mutations` | string | alpha |
| | **`--badger`** | | |
| `max_retries` | `max-retries` | int | alpha |

These are the only flag names changed. They've all been moved into their respective SuperFlags, and only `--telemetry` is new. 

**Note**: In DGRAPH-3172 it says to move `max_retries` into `--badger` "only after verifying if that flag is used only in commitToDisk." I made sure the only time the flag is used is [here](https://github.com/dgraph-io/dgraph/blob/8de50d65b99aa6a498b8fe7471ae0864dd3db6e8/worker/draft.go#L779) in the `toDisk` function, but I'm not sure if that's exactly what the ticket is referring to. 

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7544)

### CODE REVIEW

File: dgraph/cmd/alpha/run.go [hunk 1]
```diff
@@ -683,12 +683,12 @@ func run() {
 		worker.LudicrousDefaults)
 	raft := z.NewSuperFlag(Alpha.Conf.GetString("raft")).MergeAndCheckDefault(worker.RaftDefaults)
 	x.WorkerConfig = x.WorkerOptions{
+		Telemetry:           telemetry,
```
Comment (aman-bansal) [inline dgraph/cmd/alpha/run.go:686]: I don't think we needed this

File: dgraph/cmd/alpha/run.go [hunk 2]
```diff
@@ -701,9 +701,14 @@ func run() {
 		HmacSecret:          opts.HmacSecret,
 		Audit:               opts.Audit != nil,
 		Badger:              badger,
+		BadgerMaxRetries:    int(badger.GetInt64("max-retries")),
```
Comment (aman-bansal) [inline dgraph/cmd/alpha/run.go:704]: badger metrics is already there. cant we use the same super flag for this?

File: dgraph/cmd/alpha/run.go [hunk 3]
```diff
@@ -712,8 +717,6 @@ func run() {
 	setupCustomTokenizers()
 	x.Init()
 	x.Config.PortOffset = Alpha.Conf.GetInt("port_offset")
-	x.Config.Limit = z.NewSuperFlag(Alpha.Conf.GetString("limit")).MergeAndCheckDefault(
-		worker.LimitDefaults)
 	x.Config.LimitMutationsNquad = int(x.Config.Limit.GetInt64("mutations-nquad"))
 	x.Config.LimitQueryEdge = x.Config.Limit.GetUint64("query-edge")
```
Comment (aman-bansal) [inline dgraph/cmd/alpha/run.go:721]: just a quick thought wouldnt this be great to actually have struct say 
`type Limit struct {
 ...
...
}`

I am sensing with super flags we are just putting too much reliance over x.SuperFlag.Get* function and causing too much noise in various package configuration. Like limit is super flag but x.Config has all the individual limits options. Seems strange to me and bit unstructured

**Review by darkn3rd — APPROVED** [review]
:LGTM:

**Pull request merged**